python/dgl/nn/pytorch/conv/relgraphconv.py

```diff
@@ -142,10 +164,26 @@ def bdd_message_func(self, edges):
         """Message function for block-diagonal-decomposition regularizer"""
         if edges.src['h'].dtype == th.int64 and len(edges.src['h'].shape) == 1:
             raise TypeError('Block decomposition does not allow integer ID feature.')
-        weight = self.weight.index_select(0, edges.data['type']).view(
-            -1, self.submat_in, self.submat_out)
-        node = edges.src['h'].view(-1, 1, self.submat_in)
-        msg = th.bmm(node, weight).view(-1, self.out_feat)
+
+        # calculate msg @ W_r before put msg into edge
+        if self.low_mem:
```

Comment:
Can we always use the branch of `self.low_mem` is True? In other word, is the slowdown caused by `low_mem` tolerable for most cases?

Reply:
I previous test, low_mem will cause 1.5-2x slowdown for link prediction training task